app/src/main/java/mozilla/lockbox/presenter/AuthPresenter.kt

```diff
@@ -52,33 +55,38 @@ class AuthPresenter(
 
         lockedStore.onAuthentication
             .subscribe {
-                if (it is FingerprintAuthAction.OnAuthentication) {
-                    when (it.authCallback) {
-                        is FingerprintAuthCallback.OnAuth -> unlock()
-                        is FingerprintAuthCallback.OnError -> view.unlockFallback()
+                when (it) {
```

Comment:
I don't love nesting `when`s like this, it feels like a code smell to me. Ideally, we would refactor `FingerprintAuthAction` to have three cases `OnError`, `OnAuthentication`, `OnCancel` instead of needing to drill down into the `OnAuthentication` for the callback... Anyway, that's probably out of the scope of this PR unless you really feel like tackling it; otherwise I can file a follow-up issue.

Reply:
I agree it should be changed and agree that it's out of scope for the issue/PR.  Added #466 to address.